src/hooks/useDisableSpawnExpirationCheckbox.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from "@apollo/client";
+import {
+  SpawnExpirationInfoQuery,
+  SpawnExpirationInfoQueryVariables,
+} from "gql/generated/types";
+import { GET_SPAWN_EXPIRATION_INFO } from "gql/queries";
+import { MyHost, MyVolume } from "types/spawn";
+
+type ListItem = SpawnExpirationInfoQuery["myHosts"][0];
+
+const countNoExpirationCB = (accum: number, currItem: ListItem) =>
+  accum + (currItem.noExpiration ? 1 : 0);
+
+export const useDisableSpawnExpirationCheckbox = (
+  dataType: "VOLUME" | "HOST",
```

Comment:
I think a boolean prop as apposed to a ENUM would be better since there are only 2 options.
`isVolume`

Reply:
this is a good point especially since I'm only using ternaries in this component